pkg/scheduler/internal/cache/cache.go

```diff
@@ -462,6 +445,58 @@ func (cache *schedulerCache) removePod(pod *v1.Pod) error {
 	return nil
 }
 
+// Assumes that lock is already acquired.
+func (cache *schedulerCache) addPod(pod *v1.Pod, key string, isAssumedPod bool) {
+	cache.addPodToNode(pod)
+	ps := &podState{
+		pod: pod,
+	}
+	cache.podStates[key] = ps
+	if isAssumedPod {
+		cache.assumedPods.Insert(key)
+	}
+}
+
+// Assumes that lock is already acquired.
+func (cache *schedulerCache) updatePod(oldPod, newPod *v1.Pod, key string) error {
+	// removePod will delete the key from assumedPods, so we must know in advance
+	// whether this oldPod is an assumed pod.
+	isAssumedPod := cache.assumedPods.Has(key)
+
+	if err := cache.removePod(oldPod, key); err != nil {
+		return err
+	}
+
+	// A Pod delete event followed by an immediate Pod add event may be merged
+	// into a Pod update event. In this case, we should invalidate the old Pod, and
+	// then add the new Pod.
+	if oldPod.UID != newPod.UID {
+		// When the UID is different, we add the new Pod with the `newKey`
+		newKey, err := framework.GetPodKey(newPod)
+		if err != nil {
+			return err
+		}
+		key = newKey
+	}
+	cache.addPod(newPod, key, false)
```

Comment:
Doesn't `addPod` already create a new `podState`? So deadline is already nil.

Reply:
Why `isAssumed` is always false? Shouldn't we check if it exists in `assumedPods` every time?